src/lib/format/l20n/parser.js

```diff
@@ -5,201 +5,126 @@ import { L10nError } from '../../errors';
 const MAX_PLACEABLES = 100;
 
 export default {
-  _patterns: null,
-
-  init: function() {
-    this._patterns = {
-      index: /@cldr\.plural\(\$?(\w+)\)/g,
-      placeables: /\{\{\s*\$?([^\s]*?)\s*\}\}/,
-      unesc: /\\({{|u[0-9a-fA-F]{4}|.)/g,
-    };
-  },
-
-  parse: function (emit, string, simple) {
-    if (!this._patterns) {
-      this.init();
-    }
+  parse: function(env, string) {
     this._source = string;
     this._index = 0;
-    this._length = this._source.length;
-    this.simpleMode = simple;
-    this.emit = emit;
+    this._length = string.length;
+    this.entries = Object.create(null);
 
-    return this.getL20n();
+    return this.getResource();
   },
 
-  getAttributes: function() {
-    let attrs = Object.create(null);
+  getResource: function() {
+    this.getWS();
+    while (this._index < this._length) {
+      try {
+        this.getEntry();
+      } catch (e) {
+        if (e instanceof L10nError) {
+          throw e;
```

Comment:
I think we'll want to `emit` here, but if not, maybe get rid of the `try...catch`?